ouroboros-network/test/Test/Subscription.hs

```diff
@@ -597,3 +581,23 @@ _demo = ioProperty $ do
     appReq = MuxInitiatorApplication (\Mxt.ChainSync1 -> error "req fail")
     appRsp = MuxResponderApplication (\Mxt.ChainSync1 -> error "rsp fail")
 
+
+data WithThreadAndTime a = WithThreadAndTime {
+      wtatOccuredAt    :: !UTCTime
+    , wtatWithinThread :: !ThreadId
+    , wtatEvent        :: !a
+    }
```

Comment:
It seems this will be available from upstream: https://github.com/input-output-hk/iohk-monitoring-framework/pull/379/commits/fd1bbb1b6da455d68c1c74704c04d4e4b3693382
Another option is to use `ToLogObject` type class.

Reply:
Interesting, I will take a look at `threadAndTimeTracer`.